lib/instrument/transformer/nskeyed.js

```diff
@@ -202,12 +204,98 @@ class XCTCapabilities extends BaseArchiveHandler {
   }
 }
 
+/**
+ * Some of the plist classes may not be included and should not be cared.
+ * Using this as a default decoder for unknown entities.
+ */
+class NSIgnore extends BaseArchiveHandler {
+  decodeArchive() {
+    return null;
+  }
+}
+
 class NSNull extends BaseArchiveHandler {
+  static isNSObject = true;
   decodeArchive () {
     return null;
   }
 }
 
+class XCTestConfiguration extends BaseArchiveHandler {
+  static _default = {
+    aggregateStatisticsBeforeCrash: {
+      XCSuiteRecordsKey: {}
+    },
+    automationFrameworkPath: '/Developer/Library/PrivateFrameworks/XCTAutomationSupport.framework',
+    baselineFileRelativePath: undefined,
+    baselineFileURL: undefined,
+    defaultTestExecutionTimeAllowance: undefined,
+    disablePerformanceMetrics: false,
+    emitOSLogs: false,
+    formatVersion: 2,
+    gatherLocalizableStringsData: false,
+    initializeForUITesting: true,
+    maximumTestExecutionTimeAllowance: undefined,
+    productModuleName: 'WebDriverAgentRunner',
+    randomExecutionOrderingSeed: undefined,
+    reportActivities: true,
+    reportResultsToIDE: true,
+    systemAttachmentLifetime: 2,
+    targetApplicationArguments: [],
+    targetApplicationBundleID: undefined,
+    targetApplicationEnvironment: undefined,
+    targetApplicationPath: '/KEEP-THIS-NOT-EMPTY/KEEP-THIS-NOT-EMPTY',
+    testApplicationDependencies: {},
+    testApplicationUserOverrides: undefined,
+    testBundleRelativePath: undefined,
+    testExecutionOrdering: 0,
+    testTimeoutsEnabled: false,
+    testsDrivenByIDE: false,
+    testsMustRunOnMainThread: true,
+    testsToRun: undefined,
+    testsToSkip: undefined,
+    treatMissingBaselinesAsFailures: false,
+    userAttachmentLifetime: 1
+  }
+  constructor(data) {
+    super();
+    if (!data.testBundleURL) {
+      throw new Error('testBundleURL cannot be empty');
+    }
+    if (!data.sessionIdentifier) {
+      throw new Error('sessionIdentifier cannot be empty');
+    }
+    if (typeof data.testBundleURL === 'string') {
+      data.testBundleURL = new NSURL(undefined, data.testBundleURL);
+    }
+    if (!(data.testBundleURL instanceof NSURL)) {
+      throw new Error('invalid testBundleURL');
+    }
+    if (typeof data.sessionIdentifier === 'string') {
+      data.sessionIdentifier = new NSUUID(data.sessionIdentifier);
+    }
+    if (!(data.sessionIdentifier instanceof NSUUID)) {
+      throw new Error('invalid sessionIdentifier');
+    }
+    this._data = { ...XCTestConfiguration._default, ...data };
+  }
+
+  getBytes() {
+    return archive(this);
+  }
+}
+
+class XCActivityRecord extends BaseArchiveHandler {
+  static _keys = ['activityType', 'attachments', 'finish', 'start', 'title', 'uuid'];
+  decodeArchive(archive) {
+    const ret = new XCActivityRecord();
```

Comment:
I am not quite sure about what is done here. decodeArchive is already an instance method, so why do we create another instance here. Perhaps, it would make sense to just have it as a vanilla function instead?

Reply:
It should be a simple object. I must have fell asleep while doing this